server/game/gameSystems/DiscardFromDeckSystem.ts

```diff
@@ -0,0 +1,89 @@
+import { AbilityContext } from '../core/ability/AbilityContext';
+import { Card } from '../core/card/Card';
+import { EventName, GameStateChangeRequired } from '../core/Constants';
+import { IPlayerTargetSystemProperties, PlayerTargetSystem } from '../core/gameSystem/PlayerTargetSystem';
+import Player from '../core/Player';
+import { DiscardSpecificCardSystem } from './DiscardSpecificCardSystem';
+import * as Contract from '../core/utils/Contract';
+
+export interface IDiscardFromDeckProperties extends IPlayerTargetSystemProperties {
+    amount?: number;
+}
+
+export class DiscardFromDeckSystem<TContext extends AbilityContext = AbilityContext> extends PlayerTargetSystem<TContext, IDiscardFromDeckProperties> {
+    public override readonly name = 'discardFromDeck';
+    public override readonly eventName = EventName.OnDiscardFromDeck;
+
+    protected override defaultProperties: IDiscardFromDeckProperties = {
+        amount: 1
```

Comment:
I always go back and forth and whether to include default values for stuff like this. I think the question boils down to, does `AbilityHelper.immediateEffects.discardFromDeck()` intuitively read as discarding 1 card? I could genuinely go either way, just wanted to raise it